### PR TITLE
Bug 1059017 - [Calendar] RTL settings will breaks all views

### PR DESCRIPTION
**Pull request by anefzaoui**

### CODE REVIEW

**Comment by try-server-hook** [comment]
[anefzaoui Ahmed Nefzaoui (anefzaoui)](https://github.com/anefzaoui) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=47bf746b624e)

**Comment by try-server-hook** [comment]
[anefzaoui Ahmed Nefzaoui (anefzaoui)](https://github.com/anefzaoui) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=90e48ecc0481)

**Comment by try-server-hook** [comment]
[anefzaoui Ahmed Nefzaoui (anefzaoui)](https://github.com/anefzaoui) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=e023fdca4870)

File: apps/calendar/js/views/month.js [hunk 1]
```diff
@@ -72,11 +72,13 @@ Month.prototype = {
   },
 
   _nextTime: function() {
-    return monthStart(this.date, 1);
+    var isRTL =  (document.documentElement.dir==='rtl') ? true : false;
```
Comment (fabi1cazenave) [inline apps/calendar/js/views/month.js]:
```
var isRTL = (document.documentElement.dir === 'rtl');
```

File: apps/calendar/js/views/month.js [hunk 2]
```diff
@@ -72,11 +72,13 @@ Month.prototype = {
...
+    return monthStart(this.date, isRTL ? -1 : 1);
   },
 
   _previousTime: function() {
-    return monthStart(this.date, -1);
+    var isRTL =  (document.documentElement.dir==='rtl') ? true : false;
```
Comment (fabi1cazenave) [inline apps/calendar/js/views/month.js]: same here

File: apps/calendar/js/views/pan.js [hunk 3]
```diff
@@ -187,11 +192,12 @@ Pan.prototype = {
   },
 
   _set: function(x) {
+    var isRTL = (document.documentElement.dir === 'rtl') ? true : false;
```
Comment (fabi1cazenave) [inline apps/calendar/js/views/pan.js]: same here

File: apps/calendar/style/multi_day.css [hunk 4]
```diff
@@ -208,7 +208,9 @@
   border-left-width: 0.4rem;
   width: 100%;
   overflow: hidden;
-  padding: 0.6rem 0.3rem 0.6rem 0.6rem;
+  padding: 0.6rem;
+  -moz-padding-start: 0.6rem;
```
Comment (fabi1cazenave) [inline apps/calendar/style/multi_day.css]: Nit: `-moz-padding-start` is useless

File: apps/calendar/js/views/pan.js [hunk 5]
```diff
@@ -187,11 +192,12 @@ Pan.prototype = {
...
     x = clamp(x, this._minX, 0);
     this.targets.forEach(el => {
-      el.style.transform = 'translateX(' + x +'px)';
+      el.style.transform = isRTL ? 'translateX(' + -x +'px)' : 'translateX(' + x +'px)';
```
Comment (fabi1cazenave) [inline apps/calendar/js/views/pan.js]: spaces + 80-char limit

File: apps/calendar/js/views/pan.js [hunk 6]
```diff
@@ -167,7 +167,12 @@ Pan.prototype = {
     if (momentum > 0.5) {
       // if the drag was fast we consider it as a swipe (move multiple cells
       // at once)
-      var direction = this._dx > 0 ? -1 : 1;
+      var direction;
+      if (document.documentElement.dir === 'rtl') {
+        direction = this._dx > 0 ? 1 : -1;
+      } else {
+        direction = this._dx > 0 ? -1 : 1;
+      }
```
Comment (millermedeiros) [inline apps/calendar/js/views/pan.js]: can be simplified as:

```js
var direction = this._dx > 0 ? -1 : 1;
if (document.documentElement.dir === 'rtl') {
  direction *= -1;
}
```

File: apps/calendar/js/views/pan.js (hunk 5 above)
Comment (millermedeiros) [inline apps/calendar/js/views/pan.js]: I would keep old code and do:

```js
if (isRTL) {
  x *= -1;
}
```

File: apps/calendar/js/views/pan.js (hunk 3 above)
Comment (millermedeiros) [inline apps/calendar/js/views/pan.js]: maybe we should just create a helper `utils/l10n.js`:

```js
define(function(require, exports) {
'use strict';

exports.init = function() {
  update();
  window.addEventListener('localized', update);
};

exports.isRTL = null;

function update() {
  exports.isRTL = document.documentElement.dir === 'rtl';
}

});
```

I can see more methods being added in the future and that way we can keep same API even if the way l10n is handled changes in the future.

**Comment by try-server-hook** [comment]
[anefzaoui Ahmed Nefzaoui (anefzaoui)](https://github.com/anefzaoui) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=eba26aa376ff)

File: apps/calendar/js/views/pan.js [hunk 7]
```diff
@@ -37,6 +37,11 @@ function Pan(options) {
   this._onTouchEnd = this._onTouchEnd.bind(this);
   this._tick = this._tick.bind(this);
   this._onTweenEnd = null;
+
+  // RTL
+  var getDir = () => document.documentElement.dir === 'rtl' ? -1 : +1;
+  window.addEventListener('localized', () => this._dir = getDir());
+  this._dir = getDir();
```
Comment (millermedeiros) [inline apps/calendar/js/views/pan.js]: I think we should abstract this into a separate module.. we are checking for the same thing in too many places. ideally we would have something like `l10nHelper.isRTL` and add the `localized` logic inside the `l10nHelper`.

**Pull request closed without merging**